worker/caasapplicationprovisioner/application.go

```diff
@@ -210,32 +250,58 @@ func (a *appWorker) loop() error {
 		return nil
 	}
 
+	var scaleChan <-chan time.Time
+	var scaleTries int
+	var trustChan <-chan time.Time
+	var trustTries int
+	const maxRetries = 20
+	const retryDelay = 3 * time.Second
+
 	for {
 		select {
 		case _, ok := <-appScaleWatcher.Changes():
 			if !ok {
 				return fmt.Errorf("application %q scale watcher closed channel", a.name)
 			}
-			if err := a.ensureScale(app); err != nil {
-				return err
+			scaleTries = 0
+			scaleChan = time.After(0)
```

Comment:
if scale chan is not nil, no need to schedule a new one.

Reply:
Done